source/1_engine/rex_engine/src/cmdline.cpp

```diff
@@ -0,0 +1,205 @@
+#include "rex_engine/cmdline.h"
+
+#include "rex_engine/cmd_args.h"
+#include "rex_engine/diagnostics/assert.h"
+#include "rex_engine/diagnostics/logging/log_macros.h"
+#include "rex_engine/log.h"
+#include "rex_std/cstring.h"
+#include "rex_std/vector.h"
+
+namespace rex
+{
+  namespace cmdline
+  {
+    struct ActiveArgument
+    {
+      rsl::string_view argument;
+      rsl::string_view value;
+    };
+
+    class CommandLineArguments
```

Comment:
Just a question, but since we change the other class "CommandLineArgument" to cmdline::Argument. shouldn't we call this class cmdline::Arguments? Although I can see why we should keep it as this as this might become confusing with the cmdline::Argument class. One might think this is a list of this "Argument" class which it is not.

Reply:
yea, well I think this class can just be renamed to something like cmdline::ArgumentsProcessor or something